airbyte-integrations/connectors/source-postgres/src/testFixtures/java/io/airbyte/integrations/source/postgres/PostgresTestDatabase.java

```diff
@@ -36,8 +36,27 @@ public static enum BaseImage {
 
   }
 
-  static public PostgresTestDatabase in(BaseImage imageName, String... methods) {
-    final var container = new PostgresContainerFactory().shared(imageName.reference, methods);
+  public static enum ImageModifier {
+
+    ASCII("withASCII"),
+    CONF("withConf"),
+    NETWORK("withNetwork"),
+    SSL("withSSL"),
+    WAL_LEVEL_LOGICAL("withWalLevelLogical"),
+    CERT("withCert"),
+    ;
+
+    String methodName;
+
+    ImageModifier(String methodName) {
+      this.methodName = methodName;
+    }
+
```

Comment:
Suggestion: add a method `apply` (or whatever) which actually does the thing instead of naming a method in PostgresContainerFactory. You could pass a Consumer<GenericContainer<?>> to the constructor instead. This way you get rid of an unnecessary level of indirection. It's not like the Factory object will ever be stateful anyway.